src/components/molecules/PostFilter/index.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import { PageSearchParams } from "@/types/pageSearchParams";
+import { usePathname, useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+
+function PostFilter({ searchParams }: PageSearchParams) {
+  const pathname = usePathname();
+  const [searchParamsState, setSearchParamsState] = useState(new URLSearchParams(searchParams));
+  const [isAll, setIsAll] = useState(searchParamsState.get("all") !== "false");
+  const router = useRouter();
+
+  useEffect(() => {
+    setSearchParamsState(new URLSearchParams(searchParams));
+  }, [searchParams]);
+  useEffect(() => {
+    setIsAll(searchParamsState.get("all") !== "false");
+  }, [searchParamsState]);
+
+  const handleOnClick = (newIsAll: boolean) => {
+    setIsAll(newIsAll);
+    searchParamsState.set("all", newIsAll.toString());
+    const queryString = searchParamsState.toString();
+    router.push(`${pathname}?${queryString}`);
+  };
```

Comment:
위 리뷰에서 isAll, searchParamsState를 상태로 사용하지 말라고 리뷰드렸기 때문에
여기도 상태와 관련된 로직을 제거해주면 됩니다!
다만 상태를 변화시킨 것은 아니기 때문에 isAll 변수는 재평가되지 않아서 이전 값 그대로일 겁니다
하지만 router.push로 페이지 이동을 시켜주기 때문에 괜찮습니다!

```suggestion
  const handleOnClick = (newIsAll: boolean) => {
    searchParamsState.set("all", newIsAll.toString());
    const queryString = searchParamsState.toString();
    router.push(`${pathname}?${queryString}`);
  };
```